### PR TITLE
feat: Implements StringType and StringVector

### PR DESCRIPTION
**Pull request by waynexia**
Signed-off-by: Ruihang Xia <waynestxia@gmail.com>

I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?
This PR
- implements StringType and StringVector for the `datatypes2` crate (`string_type.rs` and `vectors/string.rs`)
- enables more test cases for `Value` and `ConcreteDataType`

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
#555

### CODE REVIEW

File: src/datatypes2/src/vectors/string.rs [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+    }
+}
+
+impl Serializable for StringVector {
+    fn serialize_to_json(&self) -> Result<Vec<serde_json::Value>> {
+        self.iter_data()
+            .map(|v| match v {
+                None => Ok(serde_json::Value::Null),
+                Some(vec) => serde_json::to_value(vec),
+            })
```
Comment (evenyag) [inline src/datatypes2/src/vectors/string.rs:208]: Could we use `serde_json::to_value()` directly? Maybe we need to add a test for this. BinaryVector should also use a similar pattern.
```suggestion
            .map(serde_json::to_value)
```

File: src/datatypes2/src/vectors/string.rs [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+            .collect::<serde_json::Result<_>>()
+            .context(error::SerializeSnafu)
+    }
+}
+
+vectors::impl_try_from_arrow_array_for_vector!(StringArray, StringVector);
+
+#[cfg(test)]
+mod tests {
```
Comment (evenyag) [inline src/datatypes2/src/vectors/string.rs:217]: Also ports other test cases?

File: src/datatypes2/src/vectors/string.rs [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+        let vector = StringVector::from(string_array);
+        assert_eq!(
+            r#"["A","B",null,"D"]"#,
+            serde_json::to_string(&vector.serialize_to_json().unwrap()).unwrap(),
+        );
+    }
+
+    #[test]
+    fn test_from_non_option_string() {
+        let corpus = vec!["😅😅😅", "😍😍😍😍", "🥵🥵"];
```
Comment (v0y4g3r) [inline src/datatypes2/src/vectors/string.rs:353]: You can also test if StringVector can correctly handle [NUL](https://www.compart.com/en/unicode/U+0000)  😅😅😅

**Review by v0y4g3r — APPROVED** [review]
LGTM

**Review by evenyag — APPROVED** [review]
LGTM

**Pull request merged**